src/analysis.py

```diff
@@ -298,3 +298,136 @@ def func(j_graph, start_nodes, end_nodes):
                                                                   end_station=section[1], func=func))
         flood_map.add_weighted_edges_from(ebunch_to_add=flood_map_edges)
         return flood_map
+
+    @staticmethod
+    def filter_by_gauge(graph,
+                        gauge: str
+                        ):
+        comps = list(nx.weakly_connected_components(graph))
+        edges = graph.edges()
+        comps_copy = comps.copy()
+        edges = list(edges)
+        edges_copy = edges.copy()
+
+        for comp in comps_copy:
+            if not any(gauge == elem for elem in [i[0] for i in [list(ele) for ele in list(comp)]]):
+                comps.remove(comp)
+        for i in range(len(comps)):
```

Comment:
There is a duplicated code fragments warning here. Create a function, which follows this logic and can be applied to the other methods.

Reply:
This problem occurs still in `filtering.py` (LOC81 vs. LOC147).